addons/crm_iap_lead_enrich/models/crm_iap_lead_enrich.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import datetime
+
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+from odoo.addons.iap import jsonrpc,InsufficientCreditError
+
+DEFAULT_ENDPOINT = 'https://iap-services.odoo.com'
+
+class Lead(models.Model):
+    _inherit = 'crm.lead'
+
+    enriched_lead = fields.Boolean(string='Lead enriched based on email', default=False)
+
+    @api.model
+    def lead_enrich_mail(self, cron=False):
+        domains = {}
+        for record in self:
+            if record.probability == 0:
+                continue
+            email_domain = record.partner_address_email or record.email_from
+            if email_domain:
+                domains[record.id] = email_domain.split('@')[1]
+            else:
+                record.message_post_with_view('crm_iap_lead_enrich.lead_enrich_message_no_mail', subtype_id=self.env.ref('mail.mt_note').id)        
+        if domains:
+            try:
+                response_clearbit = self._make_request(domains)
+                self._enrich_leads_from_response(response_clearbit)
+                self.env['ir.config_parameter'].sudo().set_param('lead_enrich.already_notified', False)
+
+            except InsufficientCreditError:
+                self.notify_no_more_credit('reveal', self._name, 'lead_enrich.already_notified')
+                if cron:
+                    return
+                data = {
+                    'message_title': _("Lead enriched based on email address"),
+                    'url' : self.env['iap.account'].get_credits_url('reveal'),
+                }
+                self.message_post_with_view('crm_iap_lead_enrich.lead_enrich_message_no_credit', values=data, subtype_id=self.env.ref('mail.mt_note').id)
+
+    @api.model
+    def enrich_with_cron(self):
```

Comment:
I would make this method private to avoid people calling it from outside of odoo.